main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	set "github.com/deckarep/golang-set"
+	"github.com/llgcode/draw2d/draw2dimg"
+)
+
+type LRTuple struct {
+	dir  float64
+	dist int
+}
+
+type Tuple struct {
+	rad float64
+	gx  int
+	gy  int
+}
+
+var ArrInput []LRTuple
+var xyInput []Tuple
+var iInput []Tuple
+var xyOutput []interface{}
+
+func printErr(err error) {
+	if err != nil {
+		log.Println(err.Error())
+	}
+}
+
+//parse and read textfile
+func readFile() {
+	file, err := os.Open("input.txt")
+	printErr(err)
```

Comment:
Here you print the error if there is an error and continue the function. This is a problem. If you have an error in your file opening, the rest of the function won't work!

The answer here is to return the error.

```
func readFile() error {
 file, err := os.Open("input.txt")
 if err != nil {
    return err
  }
```